base/common/python/pki/nssdb.py

```diff
@@ -183,16 +185,27 @@ def __init__(self, directory=None,
                  internal_password=None,
                  internal_password_file=None,
                  passwords=None,
-                 password_conf=None):
+                 password_conf=None,
+                 user=None,
+                 group=None):
+        self.user = user
+        self.group = group
 
         if not directory:
             directory = os.path.join(
                 os.path.expanduser("~"), '.dogtag', 'nssdb')
 
+        if user:
+            self.uid = pwd.getpwnam(user).pw_uid
+            if group:
+                self.gid = grp.getgrnam(group).gr_gid
+            else:
+                self.gid = grp.getgrnam(user).pw_gid
```

Comment:
This line probably should be like this:
```
self.gid = pwd.getpwnam(user).pw_gid
```

Reply:
You're right, there is a mistake here. It should be:
```
self.gid = pwd.getpwnam(ouser).gr_pid
```
The sequence is this:

- get the user's uid
- if a group is passed in, get the gid for it
- if not use the gid of the user